### PR TITLE
Add option to provide custom labels to tooltip and use legend labels for the tooltip value; COUNTRY=rbd

### PR DESCRIPTION
**Pull request by gislawill**
### Description
This address Feature Request #1280 with 3 updates:
1.  Allowing `<PopupContent />` (the body of the tooltip) to print one simple string rather than always `{label}: {data}`
    1. If the data provided is falsy (null, false, ""), we now just show `{label}`
    1. Current type definitions only allow null as a valid input for this use case
1. Allowing an optional `data_label` field in the layer definition that proceeds the layer's data value on the tooltip
1. Allowing an optional `display_source` field in the layer definition that uses the matching legend's label view on the tooltip

## How to test the feature:
- [ ] When running rbd instance, enable the Conflict Analysis > ACLED Conflict Index > ACLED Conflict Index for the Sahel layer
- [ ] Click on an admin region to open the tooltip

Test your changes with

- [x] `REACT_APP_COUNTRY=rbd yarn start`
- [x] `REACT_APP_COUNTRY=cambodia yarn start`
- [x] `REACT_APP_COUNTRY=mozambique yarn start`
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

## Screenshot/video of feature:
Screenshot 2024-06-26 at 2 23 11 PM

### CODE REVIEW

**Comment by gislawill** [comment]
Fixing this bug
Screenshot 2024-06-26 at 4 18 30 PM

**Comment by gislawill** [comment]
> Fixing this bug that occurs when multiple layers are selected Screenshot 2024-06-26 at 4 18 30 PM

This has been fixed

File: frontend/src/config/rbd/translation.json [hunk 1]
```diff
@@ -129,5 +129,6 @@
   "Phase classification": "Classification de la phase",
   "Population in phase 3 to 5": "Population en phase 3 à 5",
   "Days since last rain": "Les jours depuis la dernière pluie",
-  "Admin Level": "Niveau Administratif"
+  "Admin Level": "Niveau Administratif",
+  "Classification": "Classification"
```
Comment (gislawill) [inline frontend/src/config/rbd/translation.json:133]: I acknowledge this is incredibly unnecessary, it's just coverage for coverage's sake — but it demonstrates that there's coverage to everyone who doesn't already know that "Classification" = "Classification"

Comment (ericboucher) [reply frontend/src/config/rbd/translation.json:133]: Yup I think it's the right approach

**Comment by ericboucher** [comment]
@gislawill quick ask, in general I prefer to keep the titles short and explicit (ready to be merged). The link to issue is handled separately already:
Screenshot 2024-06-27 at 10 00 39 AM

I went ahead and updated the title for this PR

File: frontend/src/components/MapView/MapTooltip/PopupContent.tsx [hunk 2]
```diff
@@ -168,6 +168,10 @@ const PopupContent = ({
           return true;
         })
         .map(([key, value]) => {
+          // If the data is undefined, null, or an empty string, we don't want to show the key/value pair
```
Comment (ericboucher) [inline frontend/src/components/MapView/MapTooltip/PopupContent.tsx:171]: Can you explicit the use-case for this? Note that we already have the reverse case where we hide the `key` using the `do_not_display` prefix

File: frontend/src/components/MapView/Layers/layer-utils.tsx [hunk 3]
```diff
@@ -120,22 +143,28 @@ export const addPopupParams = (
...
-        : {}),
-      ...getFeatureInfoPropsData(
-        featureInfoPropsWithFallback || {},
-        coordinates,
-        feature,
-      ),
+      // Only if we're providing a custom label, put the data before admin level
+      ...(!useCustomLabel ? possibleAdminLevelData : {}),
+      ...featureInfoPropsData,
+      ...(useCustomLabel ? possibleAdminLevelData : {}),
```
Comment (ericboucher) [inline frontend/src/components/MapView/Layers/layer-utils.tsx:167]: Is that change expected for CH where the adminLevel ends up at the bottom?

Before:
Screenshot 2024-06-27 at 3 01 56 PM

After:
Screenshot 2024-06-27 at 3 02 01 PM

**Comment by ericboucher** [comment]
The code changes look good overall. I am a bit worried however that the tooltip logic is getting a bit complicated and might get hard to maintain / use properly

**Comment by wadhwamatic** [comment]
> The code changes look good overall. I am a bit worried however that the tooltip logic is getting a bit complicated and might get hard to maintain / use properly (from a product standpoint) cc @wadhwamatic

@ericboucher - yes, agreed. We've got a lot of configuration options happening. As I worked with it, it's pretty straightforward to use, but for someone new to it, there's a learning curve. Documentation is needed (I can contribute to that from the user perspective), but I also think renaming / simplifying things is key. Will and I touched on this yesterday. We'll create a separate issue for it

File: frontend/src/components/MapView/Layers/layer-utils.tsx (hunk 3 above)
Comment (gislawill) [reply frontend/src/components/MapView/Layers/layer-utils.tsx:167]: Good catch, I wouldn't consider this expected. I'm not certain that I know where we'd prefer "Admin Level" to go in the tooltip order.

@wadhwamatic, for tooltips showing the "IFPRI / ACLED Conflict Analysis - 2023" layer, do you have a preference for where the Admin Level is displayed?

> **Mali, Tombouctou, Tombouctou**
> IFPRI / ACLED Conflict Analysis - 2023
> _Admin Level: 2 (this is where it was prior to this change)_
> Classification: High
> Deadliness score: 84
> Deadliness rank: 47
> Danger score: 34
> Danger rank: 16
> Diffusion score: 2
> Diffusion rank: 65
> Fragmentation score: 4
> Fragmentation rank: 6
> _Admin Level: 2 (this is where's it's currently with this change)_

File: frontend/src/components/MapView/MapTooltip/PopupContent.tsx (hunk 2 above)
Comment (gislawill) [reply frontend/src/components/MapView/MapTooltip/PopupContent.tsx:171]: This allows us to pass a single string to PopupContent to display rather then always displaying a label: data. That's necessary to render the layer name on it's own line without following with `: {data}`. I'll update the comment to make that clearer — "we don't want to show the key/value pair" doesn't mean much 😅
Screenshot 2024-06-27 at 9 36 32 AM

File: frontend/src/components/MapView/Layers/layer-utils.tsx (hunk 3 above)
Comment (wadhwamatic) [reply frontend/src/components/MapView/Layers/layer-utils.tsx:167]: @gislawill - we can keep admin level at the bottom. In the issue I'll write up today, we should also address the need for this. It's not always necessary to display it.

**Review by ericboucher — APPROVED** [review]

**Comment by wadhwamatic** [comment]
@ericboucher and @gislawill - here's a new issue to start getting ideas together on the next iteration of the tooltip: https://github.com/WFP-VAM/prism-app/issues/1292

**Pull request merged**